python/BioSimSpace/Process/_atom_utils.py

```diff
@@ -0,0 +1,977 @@
+            self.lambda1 = self.protocol._getLambda1()[index]
+            self.lambda2 = self.protocol._getLambda2()[index]
+            self.alpha = self.protocol._getAlpha()[index].value()
+            self.uh = self.protocol._getUh()[index].value()
+            self.w0 = self.protocol._getW0()[index].value()
+            self.direction = self.protocol._getDirection()[index]
+            self.master_lambda = self.protocol._get_lambda_values()[index]
+        elif isinstance(
+            self.protocol, (_Protocol.AToMEquilibration, _Protocol.AToMAnnealing)
+        ):
+            self.lambda1 = self.protocol._getLambda1()
+            self.lambda2 = self.protocol._getLambda2()
+            self.alpha = self.protocol._getAlpha().value()
+            self.uh = self.protocol._getUh().value()
+            self.w0 = self.protocol._getW0().value()
+            self.direction = self.protocol._getDirection()
+
+    def _dump_atm_constants_to_dict(self):
+        """Internal function to write all ATM window-dependent constants to a dictionary (string)
+        to be used in sampling for analysis."""
+        output = ""
+        output += "atm_constants = {\n"
+        output += "    'Lambda1': {},\n".format(self.protocol._getLambda1())
+        output += "    'Lambda2': {},\n".format(self.protocol._getLambda2())
+        output += "    'Alpha': {},\n".format(
+            [i.value() for i in self.protocol._getAlpha()]
+        )
+        output += "    'Uh': {},\n".format([i.value() for i in self.protocol._getUh()])
+        output += "    'W0': {},\n".format([i.value() for i in self.protocol._getW0()])
+        output += "    'Direction': {}\n".format(self.protocol._getDirection())
+        output += "}\n"
+
+        output += "for key in atm_constants.keys():\n"
+        output += "    if key in ['Alpha','Uh','W0']:\n"
+        output += "        atm_constants[key] = [i for i in atm_constants[key] * kilocalories_per_mole]\n"
+
+        return output
+
+    def findDisplacement(self):
+        d = self.data["displacement"]
+        if isinstance(d, (list)):
+            if not all(isinstance(x, float) for x in d):
+                raise TypeError("Displacement must be a list of floats")
+            self.displacement = d
+        elif isinstance(d, _Vector):
+            disp = [d.x(), d.y(), d.z()]
+            self.displacement = disp
+        else:
+            raise TypeError("Displacement must be a list or BioSimSpace vector")
+
+    def createDisplacement(self):
+        self.findDisplacement()
+        d = [round(x, 3) for x in self.displacement]
+        output = ""
+        output += "displacement = {}\n".format(d)
+        output += "#BioSimSpace output is in angstrom, divide by 10 to convert to the expected units of nm\n"
+        output += "displacement = [i/10.0 for i in displacement]\n"
+        return output
+
+    def createSoftcorePertE(self):
+        """Create the softcorePertE function for the Gallachio lab analysis"""
```

Comment:
It's Gallicchio. This will need to be changed throughout.

Reply:
Now changed throughout. Hopefully I didn't miss any.